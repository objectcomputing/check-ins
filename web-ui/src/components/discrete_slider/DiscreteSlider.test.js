import React from "react";
import DiscreteSlider from "./DiscreteSlider";
import { mount } from "enzyme";

it("renders slider with title", () => {
  snapshot(<DiscreteSlider title="Some skill" />, {
    createNodeMock: (element) => {
      if (element.type === 'div') {
        return {
          addEventListener: jest.fn(),
        };
      }
    },
  });
});

it("value change is reported to the handler correctly", () => {
  const onChange = jest.fn();

  const wrapper = mount(
    <DiscreteSlider title="Some skill" onChange={onChange} />
  );
  const slider = wrapper.find(DiscreteSlider).at(0);
  slider.simulate("change", { target: { value: 5 } });
  slider.simulate("change", { target: { value: 4 } });
  setTimeout(() => expect(onChange.mock.calls.length).toBe(2), 0);
});

it("value change is reported to all handlers correctly", () => {
  const onChange = jest.fn();
  const onChangeCommitted = jest.fn();

  const wrapper = mount(
    <DiscreteSlider title="Some skill" onChange={onChange} onChangeCommitted={onChangeCommitted} />
  );

  const slider = wrapper.find(DiscreteSlider).at(0);
  slider.simulate("change", { target: { value: 5 } });
  slider.find("span.MuiSlider-mark").at(2).simulate("click", { target: { value: 3 } });
  setTimeout(() => {
    expect(onChange.mock.calls.length).toBe(2);
    expect(onChangeCommitted.mock.calls.length).toBe(1);
  }, 0);
});
