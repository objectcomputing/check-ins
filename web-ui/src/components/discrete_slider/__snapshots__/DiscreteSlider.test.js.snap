// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders slider with title 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  margin-bottom: 0.35em;
}

@media (max-width:899.95px) {
  .emotion-1 .MuiSlider-markLabel {
    font-size: 0.525rem;
  }
}

@media (min-width:600px) and (max-width:1199.95px) {
  .emotion-1 .MuiSlider-markLabel {
    font-size: 0.525rem;
  }
}

@media (min-width:900px) and (max-width:1535.95px) {
  .emotion-1 .MuiSlider-markLabel {
    font-size: 0.775rem;
  }
}

@media (min-width:1200px) {
  .emotion-1 .MuiSlider-markLabel {
    font-size: 0.875rem;
  }
}

<div
  className="discrete-slider"
>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom emotion-0"
    id="discrete-slider-restrict"
  >
    Some skill
  </p>
  <div
    className="emotion-1"
  >
    <input
      components={
        Object {
          "ValueLabel": [Function],
        }
      }
      data-testid="discrete-slider"
      getAriaValueText={[Function]}
      marks={
        Array [
          Object {
            "label": "Interested",
            "tooltip": "You have an interest in this skill and, perhaps, common knowledge or an understanding of its basic techniques and concepts.",
            "tooltipChildren": Array [
              "Focus on learning.",
            ],
            "value": 1,
          },
          Object {
            "label": "Novice",
            "tooltip": "You have the level of experience gained in a classroom, experimental scenarios, or via other training. You are expected to need help when performing this skill.",
            "tooltipChildren": Array [
              "Focus on developing through experience.",
              "You understand and can discuss terminology, concepts, principles and issues related to this skill.",
              "You rely heavily on reference and resource materials to be effective with this skill.",
            ],
            "value": 2,
          },
          Object {
            "label": "Intermediate",
            "tooltip": "You are able to successfully complete tasks in this competency as requested. Help from an expert may be required from time to time, but you can usually perform the skill independently.",
            "tooltipChildren": Array [
              "Focus on applying and enhancing knowledge or skill.",
              "You have applied this competency to situations, while occasionally needing minimal guidance to perform successfully.",
              "You understand and can discuss the application and implications of changes to processes and procedures in this area.",
            ],
            "value": 3,
          },
          Object {
            "label": "Advanced",
            "tooltip": "You can perform the actions associated with this skill without assistance. You are certainly recognized within your immediate organization as \\"a person to ask\\" when difficult questions arise regarding this skill.",
            "tooltipChildren": Array [
              "Focus on broad organizational/professional issues.",
              "You have consistently provided relevant ideas and perspectives on process or practice improvements as relate to this skill.",
              "You are capable of coaching others in the application of this skill.",
              "You participate in senior level discussions regarding this skill.",
              "You assist in the development of reference and resource materials related to this skill.",
            ],
            "value": 4,
          },
          Object {
            "label": "Expert",
            "tooltip": "You are known as an expert in this area. You can provide guidance, troubleshoot and answer questions related to this skill.",
            "tooltipChildren": Array [
              "Focus on the strategic.",
              "You have demonstrated consistent excellence in applying this competency across multiple projects or organizations.",
              "You are considered the “go to” person for this skill within OCI or externally.",
              "You create new applications for or lead the development of reference and resource materials for this skill.",
              "You are able to explain relevant topics, issues, process elements, and trends in sufficient detail during discussions and presentations as to foster a greater understanding among internal and external colleagues.",
            ],
            "value": 5,
          },
        ]
      }
      max={5.5}
      min={0.5}
      onChange={[Function]}
      step={null}
      type="range"
      value={3}
      valueLabelDisplay="auto"
    />
  </div>
</div>
`;
